types/service/database_service_configuration.go

```diff
@@ -226,19 +228,19 @@ type AwsRdsUsernameAndPasswordAuthConfiguration struct {
 // AwsRdsIamAuthConfiguration represents auth configuration for AWS RDS databases that use IAM authentication. You must
 // provide AWS credentials and a username. Optionally AWS CA bundle can be supplied to verify the server's certificate.
 type AwsRdsIamAuthConfiguration struct {
-	AwsCredentials
-	Username      string `json:"username"`
-	CaCertificate string `json:"ca_certificate,omitempty"`
+	AwsCredentials *common.AwsCredentials `json:"aws_credentials,omitempty"`
+	Username       string                 `json:"username"`
+	CaCertificate  string                 `json:"ca_certificate,omitempty"`
 }
 
 // GoogleCloudSqlUsernameAndPasswordAuthConfiguration represents auth configuration for Google Cloud SQL databases that
 // use username and password for authentication, and are connected to the upstream using the Cloud SQL Connector.
 // You must provide a username, a password, an Cloud SQL instance ID and Google credentials that are copied from the JSON
 // credentials file.
 type GoogleCloudSqlUsernameAndPasswordAuthConfiguration struct {
-	UsernameAndPassword
-	InstanceId         string `json:"instance_id"`
-	GcpCredentialsJson string `json:"gcp_credentials_json"`
+	UsernameAndPassword        // inherits
```

Comment:
I would prefer calling it `// embedded` or `// embedding`, because the concept of inheritance is not a thing in Go.

Reply:
removed these comments altogether.